### PR TITLE
feat(sitewide): add scroll to top button to conditionally appear in longer pages

### PR DESCRIPTION
**Pull request by DongreJaipal**
## This PR Closes Issue  closes #
https://github.com/AnguHashBlog/angular-primeng/issues/28

## Description
long blog posts there should be a scroll to pop button / icon that returns the user to the beginning of the page
for this app version primeicons used
applied in the app.component 

## What type of PR is this? (check all applicable)

- [x] 🅰️ Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Mobile & Desktop Screenshots/Recordings
[Attach screenshots or recordings if applicable]

## Added to documentation?
- [ ] 📜 README.md
- [x] 🙅 no documentation needed

## [Optional] Post-deployment tasks
[Specify any post-deployment tasks that need to be performed]

## [Optional] What gif best describes this PR or how it makes you feel? 
[Embed gif or describe the feeling in plain text]

### CODE REVIEW

File: angular-primeng-app/src/app/app.component.html [hunk 1]
```diff
@@ -3,3 +3,6 @@
   <router-outlet />
 </div>
 <app-footer />
+@if (enableScrollUp) {
```
Comment (monacodelisa) [inline angular-primeng-app/src/app/app.component.html:6]: @DongreJaipal let's name the condition `showScrollButton`

File: angular-primeng-app/src/app/app.component.html [hunk 2]
```diff
@@ -3,3 +3,6 @@
...
+<p-button class="pageUpButton" (click)="goPageTop()" icon="pi pi-angle-up" [rounded]="true" size="small"></p-button>
```
Comment (monacodelisa) [inline angular-primeng-app/src/app/app.component.html:7]: @DongreJaipal and then the on click event should be named `scrollToTop()`

File: angular-primeng-app/src/app/app.component.ts [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
+import { DOCUMENT, ViewportScroller } from "@angular/common";
 import { HeaderComponent } from "./components/header/header.component";
 import { FooterComponent } from "./components/footer/footer.component";
+import { ButtonModule } from "primeng/button";
 
 @Component({
 	selector: "app-root",
 	standalone: true,
-	imports: [RouterOutlet, HeaderComponent, FooterComponent],
+	imports: [RouterOutlet, HeaderComponent, FooterComponent,ButtonModule],
```
Comment (monacodelisa) [inline angular-primeng-app/src/app/app.component.ts:15]: @DongreJaipal could you add a space before `ButtonModule` so there is a space between the comma and the button module like this 
`imports: [RouterOutlet, HeaderComponent, FooterComponent, ButtonModule],`

File: angular-primeng-app/src/app/app.component.ts [hunk 4]
```diff
@@ -23,6 +24,9 @@ export class AppComponent implements OnInit, OnDestroy {
 	themeService: ThemeService = inject(ThemeService);
 	blogService: BlogService = inject(BlogService);
 	private querySubscription?: Subscription;
+	private scrollEvntSub?: Subscription;
+	enableScrollUp: boolean=false;
```
Comment (monacodelisa) [inline angular-primeng-app/src/app/app.component.ts:28]: @DongreJaipal for over here, I am not sure that we need this to be a subscription. 
so let's change `private scrollEvntSub?: Subscription;` to be - `showScrollButton: boolean = false;`

File: angular-primeng-app/src/app/app.component.ts [hunk 5]
```diff
@@ -50,9 +54,30 @@ export class AppComponent implements OnInit, OnDestroy {
...
+
+		this.scrollEvntSub = fromEvent(this.document, 'scroll')
+			.subscribe(() => {
+				if (this.calculatePositionScroll() > 50)
+					this.enableScrollUp = true;
+				else
+					this.enableScrollUp = false;
+			})
+	}
+
```
Comment (monacodelisa) [inline angular-primeng-app/src/app/app.component.ts:66]: @DongreJaipal and over here the using the logic you suggested, which looks good, we can use the `HostListener` which needs to be added as an import from '@angular/core' and can be used like this:

```
onWindowScroll() {
    // Check the vertical scroll position
    const yOffset = window.pageYOffset || document.documentElement.scrollTop;
    if (yOffset > 300) { // this value might need adjustments
      this.showScrollButton = true;
    } else {
      this.showScrollButton = false;
    }
  }
```
and then bellow it, the click event function 

```
  scrollToTop() {
    this.viewportScroller.scrollToPosition([0, 0]);
  }
```
please give it a try and let me know if you need any further assistance

**Review by monacodelisa — CHANGES_REQUESTED** [review]

**Review by monacodelisa — APPROVED** [review]

**Review by esteecodes — APPROVED** [review]

**Review by copypasteov — APPROVED** [review]

**Comment by monacodelisa** [comment]
@DongreJaipal thank you for your contribution 🚀

**Pull request merged**